src/providers/mbed_provider/asym_encryption.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use super::{key_management, MbedProvider};
+use crate::authenticators::ApplicationName;
+use crate::key_info_managers::KeyTriple;
+use log::info;
+use parsec_interface::operations::{psa_asymmetric_decrypt, psa_asymmetric_encrypt};
+use parsec_interface::requests::{ProviderID, ResponseStatus, Result};
+use parsec_interface::secrecy::{ExposeSecret, Secret};
+use psa_crypto::operations::asym_encryption;
+use psa_crypto::types::key;
+
+impl MbedProvider {
+    pub(super) fn psa_asymmetric_encrypt_internal(
+        &self,
+        app_name: ApplicationName,
+        op: psa_asymmetric_encrypt::Operation,
+    ) -> Result<psa_asymmetric_encrypt::Result> {
+        info!("Mbed Provider - Asym Encrypt");
+        let key_name = op.key_name.clone();
+
+        let key_triple = KeyTriple::new(app_name, ProviderID::MbedCrypto, key_name);
+        let store_handle = self.key_info_store.read().expect("Key store lock poisoned");
+        let key_id = key_management::get_key_id(&key_triple, &*store_handle)?;
+        let _guard = self
+            .key_handle_mutex
+            .lock()
+            .expect("Grabbing key handle mutex failed");
+        let id = key::Id::from_persistent_key_id(key_id);
+        let key_attributes = key::Attributes::from_key_id(id)?;
+
+        op.validate(key_attributes)?;
+        let salt_buff = match &op.salt {
+            Some(salt) => Some(salt.as_slice()),
+            None => None,
+        };
```

Comment:
I don't think that's quite it. `op.salt` is an `Option<Zeroizing<Vec<u8>>>` and I need `salt_buff` to be `Option<&[u8]>`. There may still be some more concise way of writing it that I'm not aware of though.

Reply:
Huh! I thought that since `Zeroizing` implements `Deref`, it would work - actually, it seems `as_deref` is what I should've recommended, though it doesn't solve the whole problem either - `op.salt.as_deref()` yields `Option<&Vec<u8>>`